docs/csharp/language-reference/compiler-messages/nullable-warnings.md

```diff
@@ -0,0 +1,322 @@
+
+Other situations can generate these warnings. You may have a mismatch in an interface method declaration and the implementation of that method. Or a delegate type and the expression for that delegate may differ. A type parameter and the type argument may differ in nullability.
+
+To fix these warnings, update the appropriate declaration.
+
+## Code doesn't match attribute declaration
+
+The preceding sections have discussed how you can use [Attributes for nullable static analysis](../attributes/nullable-analysis.md) to inform the compiler about the null semantics of your code. The compiler warns you if the code doesn't adhere to the promises of that attribute:
+
+- **CS8607** - *A possible null value may not be used for a type marked with `[NotNull]` or `[DisallowNull]`*
+- **CS8763** - *A method marked `[DoesNotReturn]` should not return.*
+- **CS8770** - *Method lacks `[DoesNotReturn]` annotation to match implemented or overridden member.*
+- **CS8774** - *Member must have a non-null value when exiting.*
+- **CS8776** - *Member cannot be used in this attribute.*
+- **CS8775** - *Member must have a non-null value when exiting.*
+- **CS8777** - *Parameter must have a non-null value when exiting.*
+- **CS8824** - *Parameter must have a non-null value when exiting because parameter is non-null.*
+- **CS8825** - *Return value must be non-null because parameter is non-null.*
+
+Consider the following method:
+
+:::code language="csharp" source="snippets/null-warnings/NullTests.cs" id="ViolateAttribute":::
+
+The compiler produces a warning because the `message` parameter is assigned `null` *and* the method returns `true`. The `NotNullWhen` attribute indicates that shouldn't happen.
+
+To address these warnings, update your code so it matches the expectations of the attributes you've applied. You may change the attributes, or the algorithm.
+
+## Exhaustive switch expression
+
+Switch expressions must be *exhaustive*, meaning that all input values must be handled. Even for non-nullable reference types, the `null` value must be accounted for. The compiler issues warnings when the null value isn't handled:
+
+- **CS8655** # WRN_SwitchExpressionNotExhaustiveForNull: The switch expression does not handle some null inputs (it is not exhaustive). For example, the pattern '{0}' is not covered.
+- **CS8847** # WRN_SwitchExpressionNotExhaustiveForNullWithWhen: The switch expression does not handle some null inputs (it is not exhaustive). For example, the pattern '{0}' is not covered. However, a pattern with a 'when' clause might successfully match this value.
```

Comment:
I think there was a copy paste error here. Updating to match the style of the others
```suggestion
- **CS8655** - *The switch expression does not handle some null inputs (it is not exhaustive). For example, the pattern '{0}' is not covered.*
- **CS8847** - *The switch expression does not handle some null inputs (it is not exhaustive). For example, the pattern '{0}' is not covered. However, a pattern with a 'when' clause might successfully match this value.*
```